src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -1754,6 +1785,55 @@ private void parseRecord(ConsumerRecord<String, String> record) {
         }
     }
 
+    private Optional<FlowSegmentSyncWrapperCommand> makeSyncCommand(
+            BaseFlow request, MessageContext messageContext, String replyTopic) {
+        FlowSegmentSyncWrapperCommand command;
+        if (request instanceof InstallIngressFlow) {
+            command = makeSyncCommand((InstallIngressFlow) request, messageContext, replyTopic);
+        } else if (request instanceof InstallOneSwitchFlow) {
+            command = makeSyncCommand((InstallOneSwitchFlow) request, messageContext, replyTopic);
+        } else {
+            command = null;
+        }
+        return Optional.ofNullable(command);
+    }
+
+    private FlowSegmentSyncWrapperCommand makeSyncCommand(
+            InstallIngressFlow request, MessageContext messageContext, String replyTopic) {
+        FlowEndpoint endpoint = new FlowEndpoint(
+                request.getSwitchId(), request.getInputPort(), request.getInputVlanId(), request.getInputInnerVlanId(),
+                request.isEnableLldp(), request.isEnableArp());
+        MeterConfig meterConfig = new MeterConfig(new MeterId(request.getMeterId()), request.getBandwidth());
+        FlowTransitEncapsulation encapsulation = new FlowTransitEncapsulation(
+                request.getTransitEncapsulationId(), request.getTransitEncapsulationType());
+        IngressFlowSegmentInstallCommand command = new IngressFlowSegmentInstallCommand(
+                messageContext, EMPTY_COMMAND_ID, makeSegmentMetadata(request), endpoint, meterConfig,
+                request.getEgressSwitchId(), request.getOutputPort(), encapsulation, null);
```

Comment:
Can we use `new RulesContext()` instead of `null` to do new face with NPE?

Reply:
`RulesContext()` is used only in *Remove commands and only after null check i.e it is a nullable field by its design.

As for me - invalid "default" values for a nullable field can make much more damage than null.

Fixed.